### DISCUSSION TITLE
Self-hosted runner to run jobs only if condition passes

### DISCUSSION THREAD

**Discussion by markjens** [actions/runner#1920] [Q&A] [answered]
Hi,

I'd like to use a self-hosted runner (SHR) for a public repository. I am aware of the security concerns!

My idea is to use the SHR only for `scheduled` workflows on the `main` branch, i.e. the workflow jobs should be executed **only** on approved and merged code. I know I could use `if:` in the workflow YAML, but I also want to add this check on the runner side, i.e. it should not accept any job that is not scheduled and not on the specified branch.
Is this possible ?

**Comment by henrygriffiths** [comment, marked as answer]
Hi @markjens,

Would using runner groups (https://github.com/organizations/your-org-name/settings/actions/runner-groups/new) satisfy your requirement? You can filter to a specific workflow using that feature.
